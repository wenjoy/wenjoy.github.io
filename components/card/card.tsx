import styles from './card.module.css'

export type Point = number | string

export interface Card {
  id: string
  point?: Point
}

export interface CardProps {
  point?: Point
  size?: 'small' | 'large'
  isBack?: boolean
  owner?: string
  clickHandler?: (p: Point) => void
}

export const Card = ({
  point = 0,
  size = 'small',
  isBack,
  owner,
  clickHandler = () => {},
}: CardProps) => {
  return (
    <div
      className={[styles.container, styles[size]].join(' ')}
      onClick={() => clickHandler(point)}
    >
      <div
        className={[
          styles.card,
          isBack ? styles.backface : styles.frontface,
        ].join(' ')}
        data-testid="card"
      >
        {' '}
        {isBack ? '' : point}
      </div>
      <div className={styles.owner} title={owner}>
        {owner}
      </div>
    </div>
  )
}
