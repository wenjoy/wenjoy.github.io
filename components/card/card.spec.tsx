import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Card } from './card'

describe('Card', () => {
  it('should render the card', () => {
    const { container } = render(<Card />)
    expect(container).toMatchSnapshot()
  })

  it('should render back of the card', () => {
    const { container } = render(<Card isBack />)
    expect(container).toMatchSnapshot()
    const content = screen.getByTestId('card')
    expect(content).toHaveTextContent('')
  })

  it('should display number rightly', () => {
    const { container } = render(<Card point={5} />)
    expect(container).toMatchSnapshot()
  })

  it('should display string rightly', () => {
    const { container } = render(<Card point="?" />)
    expect(container).toMatchSnapshot()
  })

  it('should display emoji rightly', () => {
    const { container } = render(<Card point="â˜•" />)
    expect(container).toMatchSnapshot()
  })

  it('should call clickHandler prop when clicked', () => {
    const handleClick = jest.fn()
    render(<Card clickHandler={handleClick} point={1} />)
    fireEvent.click(screen.getByText('1'))
    expect(handleClick).toHaveBeenCalledWith(1)
  })
})
