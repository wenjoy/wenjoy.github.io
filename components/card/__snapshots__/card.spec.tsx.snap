// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should display emoji rightly 1`] = `
<div>
  <div
    class="container small"
  >
    <div
      class="card frontface"
      data-testid="card"
    >
       
      â˜•
    </div>
    <div
      class="owner"
    />
  </div>
</div>
`;

exports[`Card should display number rightly 1`] = `
<div>
  <div
    class="container small"
  >
    <div
      class="card frontface"
      data-testid="card"
    >
       
      5
    </div>
    <div
      class="owner"
    />
  </div>
</div>
`;

exports[`Card should display string rightly 1`] = `
<div>
  <div
    class="container small"
  >
    <div
      class="card frontface"
      data-testid="card"
    >
       
      ?
    </div>
    <div
      class="owner"
    />
  </div>
</div>
`;

exports[`Card should render back of the card 1`] = `
<div>
  <div
    class="container small"
  >
    <div
      class="card backface"
      data-testid="card"
    >
       
    </div>
    <div
      class="owner"
    />
  </div>
</div>
`;

exports[`Card should render the card 1`] = `
<div>
  <div
    class="container small"
  >
    <div
      class="card frontface"
      data-testid="card"
    >
       
      0
    </div>
    <div
      class="owner"
    />
  </div>
</div>
`;
